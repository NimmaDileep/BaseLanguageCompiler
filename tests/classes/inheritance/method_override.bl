class Vehicle {
    var model: String;
    var speed: Int;

    fun drive() -> Unit {
        var desc: String = concat("driving at ", str(self.speed));

        print(desc);
    }
}

class Car: Vehicle {
    var year: Int;
    

    fun drive() -> Unit {
        var desc1: String = concat("driving at ", str(self.speed));
        var desc2: String = concat(" in ", str(self.year));

        var desc: String = concat(desc1, desc2);

        print(desc);
    }
}


var vehicles: [Vehicle] = [
    Vehicle("Boeing", 200),
    Vehicle("Cessna", 150),
    Car("Tesla", 200, 2019)
];

for (var i: Int = 0; i < len(vehicles); i = i + 1) {
    var vehicle: Vehicle = vehicles[i];

    vehicle.drive();
}
