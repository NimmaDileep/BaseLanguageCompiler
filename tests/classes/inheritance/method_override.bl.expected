driving at 200
driving at 150
driving at 200 in 2019
